test/mocks.ts

```diff
@@ -8,3 +20,40 @@ export function createMockEnvironment(): jest.Mocked<Environment> {
         startApplication: jest.fn<Promise<void>, [StoredApplication]>()
     };
 }
+
+export function createMockApiHandler(): jest.Mocked<APIHandler<APITopic, Events>> {
+    const mockApiHandler = new APIHandler<APITopic, Events>() as jest.Mocked<APIHandler<APITopic, Events>>;
+
+    (mockApiHandler as PartiallyWritable<typeof mockApiHandler, 'onConnection'>).onConnection = new Signal<[Identity]>();
+    (mockApiHandler as PartiallyWritable<typeof mockApiHandler, 'onDisconnection'>).onDisconnection = new Signal<[Identity]>();
+
+    return mockApiHandler;
+}
+
+export function createMockClientRegistry(): jest.Mocked<ClientRegistry> {
+    const mockClientRegistry = new ClientRegistry(null!, null!, null!) as jest.Mocked<ClientRegistry>;
+    (mockClientRegistry as PartiallyWritable<typeof mockClientRegistry, 'onAppActionReady'>).onAppActionReady = new Signal<[Identity]>();
+
+    return mockClientRegistry;
+}
+
+export function createMockDatabase(): jest.Mocked<Database> {
+    return new Database() as jest.Mocked<Database>;
+}
+
+export function createMockFin(): jest.Mocked<Fin> {
```

Comment:
How re-usable would this be? I imagine every time we want a mock of `fin`, we'd want to mock a different subset of it - and merging all of those into a single mock (which is scriptable through tests) would be a rather large undertaking.

Reply:
As discussed, will merge this into SERVICE-593